llama_cpp/src/Utility/DataGenerator.h

```diff
@@ -35,11 +37,28 @@ template<>
 void RandomData::generateRandomData<int>(std::vector<int> *targetVector, int count);
 
 template<>
-void RandomData::generateRandomData<bool>(std::vector<bool> *targetVector, int count);
+void RandomData::generateRandomData<double>(std::vector<double> *targetVector, int count);
 
 template<>
-void RandomData::generateRandomData<double>(std::vector<double> *targetVector, int count);
+void RandomData::generateRandomData<std::string>(std::vector<std::string> *targetVector, int count);
+
+class ContinuousData {
+ public:
+    ContinuousData() {}
+
+    template<typename T>
+    void generateContinuousData(std::vector<T> *targetVector, int count, bool reverse);
+};
+
+template<>
+void ContinuousData::generateContinuousData<int>(std::vector<int> *targetVector, int count, bool reverse);
+
+template<>
+void ContinuousData::generateContinuousData<int>(std::vector<int> *targetVector, int count, bool reverse);
+
+template<>
+void ContinuousData::generateContinuousData<int>(std::vector<int> *targetVector, int count, bool reverse);
```

Comment:
These can be eliminated. Besides, you have three int template specialisations here.